plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3select/TestS3SelectRecordCursorProvider.java

```diff
@@ -0,0 +1,158 @@
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMNS;
+import static org.apache.hadoop.hive.serde.serdeConstants.LIST_COLUMN_TYPES;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_LIB;
+import static org.testng.Assert.assertTrue;
+
+public class TestS3SelectRecordCursorProvider
+{
+    private static final Configuration CONFIGURATION = ConfigurationInstantiator.newEmptyConfiguration();
+    private static final Path PATH = new Path("s3://fakeBucket/fakeObject.gz");
+    private static final long START = 0;
+    private static final long LENGTH = 10;
+    private static final long FILESIZE = 10;
+    private static final HdfsEnvironment HDFS_ENVIRONMENT = new TestingHdfsEnvironment(new ArrayList<>());
+    private static final S3SelectRecordCursorProvider S3_SELECT_RECORD_CURSOR_PROVIDER = new S3SelectRecordCursorProvider(HDFS_ENVIRONMENT, new TrinoS3ClientFactory(new HiveConfig()));
+    private static boolean s3SelectPushdownEnabled = true;
+    private static final List<HiveColumnHandle> SCHEMA_COLUMNS = ImmutableList.of(ARTICLE_COLUMN, AUTHOR_COLUMN, DATE_ARTICLE_COLUMN, QUANTITY_COLUMN);
+    private static final Properties SCHEMA = createTestingSchema();
+
+    @Test
+    public void shouldReturnSelectRecordCursor()
+    {
+        List<HiveColumnHandle> columnHandleList = new ArrayList<>();
+        s3SelectPushdownEnabled = true;
+        TupleDomain<HiveColumnHandle> effectivePredicate = TupleDomain.all();
+        Optional<HiveRecordCursorProvider.ReaderRecordCursorWithProjections> recordCursor =
+                S3_SELECT_RECORD_CURSOR_PROVIDER.createRecordCursor(
+                        CONFIGURATION, SESSION, PATH, START, LENGTH, FILESIZE, SCHEMA, columnHandleList, effectivePredicate, TESTING_TYPE_MANAGER, s3SelectPushdownEnabled);
+        assertTrue(recordCursor.isPresent());
+    }
+
+    @Test
+    public void shouldReturnSelectRecordCursorWhenEffectivePredicateExists()
+    {
+        s3SelectPushdownEnabled = true;
+        TupleDomain<HiveColumnHandle> effectivePredicate = withColumnDomains(ImmutableMap.of(QUANTITY_COLUMN,
+                Domain.create(SortedRangeSet.copyOf(BIGINT, ImmutableList.of(Range.equal(BIGINT, 3L))), false)));
+        Optional<HiveRecordCursorProvider.ReaderRecordCursorWithProjections> recordCursor =
+                S3_SELECT_RECORD_CURSOR_PROVIDER.createRecordCursor(
+                        CONFIGURATION, SESSION, PATH, START, LENGTH, FILESIZE, SCHEMA, SCHEMA_COLUMNS, effectivePredicate, TESTING_TYPE_MANAGER, s3SelectPushdownEnabled);
+        assertTrue(recordCursor.isPresent());
+    }
+
+    @Test
+    public void shouldReturnSelectRecordCursorWhenProjectionExists()
+    {
+        s3SelectPushdownEnabled = true;
+        TupleDomain<HiveColumnHandle> effectivePredicate = TupleDomain.all();
+        final List<HiveColumnHandle> readerColumns = ImmutableList.of(QUANTITY_COLUMN, AUTHOR_COLUMN, ARTICLE_COLUMN);
```

Comment:
nit: drop `final` (here an in `shouldNotReturnSelectRecordCursorWhenProjectionOrderIsDifferent`)

Reply:
Thanks. Fixing it in second revision.